src/main/java/cn/edu/tsinghua/tsfile/timeseries/read/RecordReader.java

```diff
@@ -44,7 +51,14 @@ public RecordReader(ITsRandomAccessFileReader raf) throws IOException {
      */
     public DynamicOneColumnData getValueInOneColumn(DynamicOneColumnData res, int fetchSize
             , String deltaObjectUID, String measurementUID) throws IOException {
-        checkSeries(deltaObjectUID, measurementUID);
+        try {
+            checkSeries(deltaObjectUID, measurementUID);
+        }catch(IOException ex){
+            if(res == null)res = new DynamicOneColumnData();
+            res.dataType = fileReader.getDataTypeBySeriesName(deltaObjectUID, measurementUID);
```

Comment:
The invoking of ```getDataTypeBySeriesName``` is wrong.
In ```getDataTypeBySeriesName``` method of ```FileReader```, ```getRowGroupReaderMap``` is invoked, and in ```getRowGroupReaderMap``` method, ```loadAllDeltaObj();``` is invoked, we just need read one series to get the data type of a ```deltaObjectUID.measurementUID```